### ISSUE TITLE
Node Creation dialog can be invoked, even though constraints don't allow any NodeTypes to be inserted

### ISSUE THREAD

**Issue by grebaldi** [neos/neos-ui#2929]
### Description

Given there's a content collection on the page that does not allow any content to be inserted, it is still possible to invoke the node creation dialog. This is a minor issue, since it doesn't make much sense to have such a content collection in the first place, but nevertheless, the "Add Node" button should be disabled in this scenario.

This behavior is possibly related to  #2917.

### Steps to Reproduce

1. Render a content collection on the page that doesn't allow any NodeTypes to be inserted
2. Select that content collection in Neos UI
3. You'll notice that the "Add Node" button can be clicked.
4. When you click on the button, an empty NodeCreationDialog shows up.

#### Expected behavior

The "Add Node" button is disabled when the selected collection does not allow for any NodeTypes to be inserted.

#### Actual behavior

The "Add Node" button is enabled and leads to an empty NodeCreationDialog.

![Screenshot_2021-07-25_13-57-09](https://user-images.githubusercontent.com/2522299/126902962-105f07aa-e46b-4ddd-ad76-fb21dfc01226.png)

![Screenshot_2021-07-25_13-56-39](https://user-images.githubusercontent.com/2522299/126902967-e428a36f-f85f-433e-a8bb-92e0645a6dde.png)

### Affected Versions

Neos: 5.3

UI: 5.3
